test/old_tests/UnitTests/References.cpp

```diff
@@ -27,4 +28,44 @@ TEST_CASE("References")
 
     // Tests short circuit when one is empty.
     REQUIRE(a != e);
+
+    // Different integer types, but same value
+    REQUIRE(c == f);
+
+    // Different integer types, but one is null.
+    REQUIRE(e != f);
+
+    // Tests comparison to value.
+    REQUIRE(a == 123);
+    REQUIRE(123 == a);
+
+    REQUIRE(a != 124);
+    REQUIRE(124 != a);
+
+    // Tests short circuit when reference is empty.
+    REQUIRE_FALSE(e == 123);
+    REQUIRE_FALSE(123 == e);
+
+    REQUIRE(e != 123);
+    REQUIRE(123 != e);
+}
+
+TEST_CASE("String References")
+{
+    // Tests comparison to string literal
+    IReference<hstring> str{ L"foo" };
+    IReference<hstring> nullstr{ nullptr };
+
+    REQUIRE(str == L"foo");
+    REQUIRE(L"foo" == str);
+
+    REQUIRE(str != L"bar");
+    REQUIRE(L"bar" != str);
+
+    // Tests short circuit when reference is empty.
+    REQUIRE_FALSE(nullstr == L"foo");
+    REQUIRE_FALSE(L"foo" == nullstr);
+
+    REQUIRE(nullstr != L"bar");
+    REQUIRE(L"bar" != nullstr);
```

Comment:
Test these combinations also:

- IReference<hstring> and hstring
- IReference<hstring> and std::wstring
- IReference<hstring> and std::wstring_view
- IReference<hstring> and char[] (non-literal)
- IReference<hstring> and char* (non-literal)